GithubHandlers.py

```diff
@@ -152,29 +164,54 @@ def get_file_contents(self, filepath: str, ref=None):
             traceback.print_exc()
             return "Could not retrieve file contents."
 
-    def get_pr_deltas(self, pr):
+    def get_pr_deltas(self, pr) -> str:
         """
         Gets the deltas for a pull request.
 
         Args:
             pr (PullRequest): The pull request to get deltas for.
 
         Returns:
-            str: The deltas for the pull request.
+            str: The deltas for the pull request plus additional code context.
         """
-
-        valid_extensions = self.FILE_EXTENSIONS
-        files_and_deltas = ""
+        self.index_pr(pr)
+        delta_query = ""
         file_changes = pr.get_files()
         for f in file_changes:
-            if not any([f.filename.endswith(ext) for ext in valid_extensions]):
+            if not any([f.filename.endswith(ext) for ext in self.FILE_EXTENSIONS]):
                 continue
-            file_full_content = self.get_file_contents(f.filename, ref=pr.head.sha)
-            full_file_delta = self.get_full_file_delta(f.filename, file_full_content)
+
             file_delta_str = self.get_file_delta(f.patch, f.filename)
-            files_and_deltas += full_file_delta + "\n" + file_delta_str + "\n"
+            file_delta_context = self.get_file_context(file_delta_str)
+            delta_query_block = self.get_delta_query_block(
+                file_delta_str, file_delta_context
+            )
+            delta_query += delta_query_block
+
+        return delta_query
+
+    def get_delta_query_block(self, file_delta_str, file_delta_context):
+        query_block = f"""Add a code review comment to the following changes:
+        {file_delta_str}
+        
+        Only for context, here is some code from the repository (DO NOT REVIEW THIS CODE, IT IS ONLY FOR CONTEXTUAL INPUT):
+        {file_delta_context}\n"""
+        return query_block
+
```

Comment:
The method `get_file_context` is adding context to the code reviews, but ensure that this additional context is optimized for both relevance and performance.